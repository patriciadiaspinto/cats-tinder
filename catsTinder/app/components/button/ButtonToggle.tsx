/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import type {PropsWithChildren} from 'react';
import {Platform, StyleSheet, TouchableOpacity, View} from 'react-native';
import {Star} from '../icons';

type ButtonToggleProps = PropsWithChildren<{
  id: string;
  onPress: () => void;
  checked: boolean;
}>;

const ButtonToggle = ({id, onPress, checked, children}: ButtonToggleProps) => {
  const justifyContentStyle = checked ? 'flex-start' : 'flex-end';
  return (
    <View style={{...styles.toggle, justifyContent: justifyContentStyle}}>
      <TouchableOpacity
        id={id}
        onPress={onPress}
        testID={id}
        style={styles.button}>
        {children || null}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    width: 40,
    height: 24,
    borderRadius: 15,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    paddingVertical: 4,
    paddingHorizontal: 12,
  },
  toggle: {
    padding: 2,
    display: 'flex',
    flexDirection: 'row',
    width: 84,
    height: 28,
    borderRadius: 28,
    backgroundColor: '#E3E3E4',
    alignItems: 'center',
    margin: 20,
  },
});

export default ButtonToggle;
