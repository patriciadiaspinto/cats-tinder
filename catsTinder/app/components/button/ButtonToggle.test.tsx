import {act, fireEvent, render} from '@testing-library/react-native';
import ButtonToggle from './ButtonToggle';

describe('Button Toggle Component', () => {
  const defaultProps = {
    id: 'button-toggle',
    onPress: jest.fn(),
    checked: true,
  };

  it(' should render Button Icon', () => {
    const {getByTestId} = render(
      <ButtonToggle
        id={defaultProps.id}
        onPress={defaultProps.onPress}
        checked={defaultProps.checked}
      />,
    );

    expect(getByTestId('button-toggle')).toBeDefined;

    const tree = render(
      <ButtonToggle
        id={defaultProps.id}
        onPress={defaultProps.onPress}
        checked={defaultProps.checked}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('onPress', async () => {
    const onPressSpy = jest.spyOn(defaultProps, 'onPress');
    const {getByTestId} = render(
      <ButtonToggle
        id={defaultProps.id}
        onPress={defaultProps.onPress}
        checked={defaultProps.checked}
      />,
    );

    const btnIcon = getByTestId('button-toggle');
    await act(async () => {
      fireEvent.press(btnIcon);
    });

    expect(onPressSpy).toHaveBeenCalled();
  });
});
