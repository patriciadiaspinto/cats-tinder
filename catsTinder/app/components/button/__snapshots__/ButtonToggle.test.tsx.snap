// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Toggle Component  should render Button Icon 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#E3E3E4",
      "borderRadius": 28,
      "display": "flex",
      "flexDirection": "row",
      "height": 28,
      "justifyContent": "flex-start",
      "margin": 20,
      "padding": 2,
      "width": 84,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 15,
        "display": "flex",
        "height": 24,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "width": 40,
      }
    }
    testID="button-toggle"
  />
</View>
`;
